src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -54,21 +72,34 @@ boot_comp_pgtbl_expand(size_t n_pte, pgtblcap_t pt, vaddr_t vaddr, struct cobj_h
 	}
 
 	for (i = 0; i < n_pte; i++) {
-		if (!cos_pgtbl_intern_alloc(&boot_info, pt, vaddr, SERVICE_SIZE)) BUG();
+		if (!cos_pgtbl_intern_alloc(boot_info, pt, vaddr, SERVICE_SIZE)) BUG();
 	}
 }
 
+#define RESMGR_UNTYPED_MEM_SZ (COS_MEM_KERN_PA_SZ - (4 * PGD_RANGE))
```

Comment:
should we make this dynamic?  This just seems like something that will break when we try and load something that requires more than 16 MB of memory.  Not sure what's best here.

Reply:
Yeah, I agree:
1. In the current implementation, llbooter allocates this SZ to RESMGR when it's loading RESMGR. That will definitely make booter not be able to load remaining objects together cross 16MB.. At the very least, it should alloc to resmgr at the end of loading all components.
2. Like you said, we can make this dynamic at the end of loading all components based on the remaining untyped with booter!

I'll look into making that change!